pkg/util/sys/device.go

```diff
@@ -399,35 +411,106 @@ func parseUdevInfo(output string) map[string]string {
 	return result
 }
 
-func isDeviceAvailable(executor exec.Executor, devicePath string) (bool, string, error) {
-	CVInventory, err := inventoryDevice(executor, devicePath)
+// if this function returns an error, the disk should not be used (should be considered unavailable)
+func rejectDeviceIfBluestore(devicePath string) (available bool, rejectedReason string, outErr error) {
+	if SkipDeviceOpenForUnitTests {
+		logger.Infof("%s: %q", SkipDeviceOpenForUnitTestsInfoMessage, devicePath)
+		// if this is a unit test, just return that the device is available to avoid making calls to
+		// open a disk that might not exist on the system. This function was tested manually on a
+		// system with a mix of bluestore and non-bluestore disks and should be tested whenever
+		// there are changes in the future.
+		return true, "", nil
+	}
+
+	// device must not be locked - try to open in read-write exclusive mode to test if it's locked or not
+	// related: https://github.com/ceph/ceph/pull/24504
+	f, err := os.OpenFile(devicePath, os.O_RDWR|os.O_EXCL, os.FileMode(0))
```

Comment:
This function opens a raw device (e.g., /dev/sda) which I don't think we are able to mock. I implemented this with a boolean `SkipDeviceOpenForUnitTests` we can set to true for unit tests, and I verify that the message `SkipDeviceOpenForUnitTestsInfoMessage` (indicates this check is being skipped) doesn't appear in the osd-prepare pod logs in the integration smoke test at runtime.

@leseb @travisn let me know if you have any concerns or other suggestions.

Reply:
Have you seen https://talks.golang.org/2012/10things.slide#8 to mock filesystem? But if this becomes troublesome, I'm fine skipping this with a variable in the test.

Also, my previous question was left unanswered, so here it is again :) 
Should we use `os.ModeDevice()`  instead of `os.FileMode(0)`? Feels more appropriate.

Reply:
os.ModeDevice() is not equivalent to os.FileMode(0), and mode 0 is what was used in ceph-volume's code. You mentioned Erwan wrote that and that you trust his implementation.